### DISCUSSION TITLE
HTTPS?: Form submission is not possible

### DISCUSSION THREAD

**Discussion by InTerestedOWL** [statamic/cms#6877] [Troubleshooting]
Hey there,
I am currently building a website with statamic. 

I am now facing some issues regarding the use of https and the form submission:

For explanation I have a lokal environment and a live environment on a server.

My problem is the following: While I can submit the form on the local environment, my live environment reports me a fatal PHP Error which I can't really understand:

```
Error
Call to a member function lang() on null
```

I am not new to the game. I looked after the reason: 

And found the FormController.php line 51 which is causing the issue:
` $this->withLocale($site->lang(), function () use ($fields) {`

Checked the $site variable it is obvious in line 34 that there is something wrong. I debugged a bit:

` $site = Site::findByUrl(URL::previous());`
 
 While URL::previous() returns the previous page, which behaves as expected. 
 But Site::findByUrl() returns null. Which is the problem..
 
 Since css and co was not loaded in the control panel via https. I added URL::forceSchema('https'); to the web.php channel. Maybe this is a fault?

**Comment by jasonvarga** [comment]
Can you show your config/statamic/sites.php file?

**Reply by InTerestedOWL** [reply]
Didnt' changed it at all:

```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Sites
    |--------------------------------------------------------------------------
    |
    | Each site should have root URL that is either relative or absolute. Sites
    | are typically used for localization (eg. English/French) but may also
    | be used for related content (eg. different franchise locations).
    |
    */

    'sites' => [

        'default' => [
            'name' => config('app.name'),
            'locale' => 'en_US',
            'url' => '/',
        ],

    ],
];
```

**Reply by InTerestedOWL** [reply]
I also tried to change it to env('APP_URL'). 

.env sets APP_URL to my production URL starting with https://

**Reply by jasonvarga** [reply]
Instead of `'url' => '/',` try doing `'url' => env('APP_URL')` so it uses the https.

**Reply by InTerestedOWL** [reply]
I tried it, does not work

**Reply by jasonvarga** [reply]
Is your config cached?

**Reply by InTerestedOWL** [reply]
Yeah I cleared it already. Every page does then respond with a 404, except for the control panel. That does not help me, right a way.

I tried then to remove the URL::forceSchema('https') in routes/web.php, but this follows no styling within the customer panel.

**Reply by jasonvarga** [reply]
Are you using cloudflare?

**Reply by InTerestedOWL** [reply]
I do not use cloudflare at all. The page is available freely.

Everything works as expected except for the form submission which fails as I mentioned in the FormController.

**Reply by jasonvarga** [reply]
What service have you used to get an SSL certificate?

**Reply by InTerestedOWL** [reply]
The ssl  certificate is from Let's Encrypt

**Reply by InTerestedOWL** [reply]
I don't think, that we have to search the problem at the ssl certificate. I don't understand, why Site::findByUrl('https://:example.com/site') is returning null? 

What have I todo that the complete page is rendered correctly over https without 404 errors and fatal php errors caused by failed finds of my URL ....

**Reply by InTerestedOWL** [reply]
Tested it to submit the formular. over the live preview from the control panel. 

And surprise: There it is working correctly! It is submitted without error?! Could not be the reason. I suppose the error in the configuration of the https.
